### PR TITLE
REST API: make extra-program-pages and apps-total-extra-pages optional

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

Make extra-program-pages and apps-total-extra-pages optional in order to do not exposing them before the protocol switch.

## Test Plan

Existing tests updated

### CODE REVIEW

File: daemon/algod/api/server/v2/account.go [hunk 1]
```diff
@@ -125,8 +125,13 @@ func AccountDataToAccount(
 		AuthAddr:                    addrOrNil(record.AuthAddr),
 		AppsLocalState:              &appsLocalState,
 		AppsTotalSchema:             &totalAppSchema,
-		AppsTotalExtraPages:         &totalExtraPages,
```
Comment (tsachiherman) [inline daemon/algod/api/server/v2/account.go:128]: you could have just called
```golang
AppsTotalExtraPages:         numOrNil(totalExtraPages),
```

Comment (algorandskiy) [reply daemon/algod/api/server/v2/account.go:128]: thanks, did not know about this helper, fixed!

**Review by tsachiherman — APPROVED** [review]
looks good.

**Review by jasonpaulos — APPROVED** [review]
Looks good

**Comment by winder** [comment]
Is this really something people are asking for?

**Comment by jasonpaulos** [comment]
Not really an answer, but I'm in favor of making all "optional" API fields actually optional, and this is a step in that direction.

**Comment by winder** [comment]
> Not really an answer to the above question, but I'm in favor of making all "optional" API fields actually optional, and this is a step in that direction.

Yes, that's fine, but I would probably accommodate that by marking this field as required in the REST API spec.

**Comment by algorandskiy** [comment]
> Yes, that's fine, but I would probably accommodate that by marking this field as required in the REST API spec.

Please follow indexer discussion. We want to have this field not set before protocol upgrade.

**Pull request merged**